app/ydoc-shared/src/ast/tree.ts

```diff
@@ -2691,14 +2694,31 @@ export class MutableFunctionDef extends FunctionDef implements MutableStatement
   spliceArgumentDefinitions(
     start: number,
     deletedCount: number,
-    ...newValues: OwnedArgumentDefinitions[]
+    ...newValues: OwnedArgumentDefinition[]
   ) {
     const defs = [...this.fields.get('argumentDefinitions')]
     const newDefs = newValues.map((def) => mapRefs(def, ownedToRaw(this.module, this.id)))
     defs.splice(start, deletedCount, ...newDefs)
     this.fields.set('argumentDefinitions', defs)
   }
 
+  setArgumentType(index: number, typeExpr: Owned<MutableExpression> | undefined) {
+    const defs = [...this.fields.get('argumentDefinitions')]
+    if (defs.length > index) {
+      const def = defs[index]!
+      const type: ArgumentType | undefined =
+        typeExpr ?
+          {
+            operator:
+              def.type?.operator ?? unspaced(Token.new(':', TokenType.TypeAnnotationOperator)),
+            type: concreteChild(this.module, unspaced(typeExpr), this.id),
```

Comment:
`unspaced(typeExpr)` looks wrong if there is a previously-existing `def.type` and we reuse its `operator`, which may be spaced.